### PR TITLE
Use native support for CUDA in CMake not FindCUDA macros

### PR DESCRIPTION
**Pull request by MarcusGDaniels**
…ption of cuda_select_arch_flags, to using the native CMake support for CUDA.  This makes it possible to compile just qv_avx2.cpp with -mfma and -mavx2.

### Summary

This change addresses the problem in issue #843 where all files are built with -mfma and -mavx2 which creates incompatible binaries for some systems.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -249,30 +244,40 @@ else() # Standalone build
...
-	endif()
-	set(AER_SIMULATOR_SOURCES "${PROJECT_SOURCE_DIR}/contrib/standalone/qasm_simulator.cpp"
-		"${SIMD_SOURCE_FILE}")
-	set_source_files_properties(${AER_SIMULATOR_SOURCES} PROPERTIES CUDA_SOURCE_PROPERTY_FORMAT OBJ)
 
+	set(AER_SIMULATOR_SOURCES "${PROJECT_SOURCE_DIR}/contrib/standalone/qasm_simulator.cpp")
 	set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 	if(CUDA_FOUND)
-		cuda_add_executable(qasm_simulator ${AER_SIMULATOR_SOURCES})
+		set_source_files_properties(${SIMD_SOURCE_FILE} PROPERTIES LANGUAGE CUDA)
```
Comment (atilag) [inline CMakeLists.txt:251]: hmmm this is somhow a special file, not sure what would imply to set the `LANGUAGE` to `CUDA` here, but we must be extra careful with everything involving this file.

**Comment by atilag** [comment]
Thanks for taking over this issue @MarcusGDaniels !
It's not going to be trivial, but definitely really helpful.

File: CMakeLists.txt [hunk 2]
```diff
@@ -8,9 +8,9 @@
 # For Mac, statically linking only happens with user libraries, system libraries cannot
 # be linked statically per Apple's indications.
 
-cmake_minimum_required(VERSION 3.6)
+cmake_minimum_required(VERSION 3.8 FATAL_ERROR)
```
[the remark this answers is not in the document]
Comment (MarcusGDaniels) [reply CMakeLists.txt:11]: Yeah, the version change is to use [this](https://cmake.org/cmake/help/v3.8/release/3.8.html#id5) and the FATAL_ERROR to [force old versions](https://cmake.org/cmake/help/latest/command/cmake_minimum_required.html) of CMake to do it.

File: CMakeLists.txt (hunk 1 above)
Comment (MarcusGDaniels) [reply CMakeLists.txt:251]: This is the new general way to do the same as `set_source_files_properties(.. PROPERTIES CUDA_SOURCE_PROPERTY_FORMAT OBJ)` (how FindCUDA did it).

File: CMakeLists.txt [hunk 3]
```diff
@@ -192,12 +192,7 @@ if(AER_THRUST_SUPPORTED)
 			endif()
 		endif()
 		cuda_select_nvcc_arch_flags(AER_CUDA_ARCH_FLAGS ${AER_CUDA_ARCH})
-		set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS} -ccbin ${CMAKE_CXX_COMPILER} ${AER_CUDA_ARCH_FLAGS} -DAER_THRUST_CUDA -std=c++14 -I${AER_SIMULATOR_CPP_SRC_DIR} -isystem ${AER_SIMULATOR_CPP_SRC_DIR}/third-party/headers -use_fast_math --expt-extended-lambda")
```
[the remark this answers is not in the document]
Comment (MarcusGDaniels) [reply CMakeLists.txt:195]: ok, I'll update my branch to use the new conventions here too.

[the remark this answers is not in the document]
Comment (MarcusGDaniels) [reply CMakeLists.txt:195]: did that

**Comment by MarcusGDaniels** [comment]
Is there a version of CMake that is preferred?   I notice that the Python build uses its own, so differences between the host CMake and the Python version can create confusing situations.  One example is whether to use FindCUDA to locate Cuda or not.  Only the latest versions of CMake (3.17) have the new macro, but relatively old versions of CMake have the native CUDA support (3.8).

**Pull request closed without merging**